File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/events/LibraryChosenEventReceiver.java
package com.lasthopesoftware.bluewater.client.browsing.library.events;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.lasthopesoftware.bluewater.client.browsing.BrowserEntryActivity;
import com.lasthopesoftware.bluewater.settings.repository.ApplicationConstants;

public class LibraryChosenEventReceiver extends BroadcastReceiver {

	private final BrowserEntryActivity browserEntryActivity;

	public LibraryChosenEventReceiver(BrowserEntryActivity browserEntryActivity) {
		this.browserEntryActivity = browserEntryActivity;
	}

	@Override
	public void onReceive(Context context, Intent intent) {
		final int chosenLibrary = intent.getIntExtra(ApplicationConstants.PreferenceConstants.chosenLibraryKey, -1);
		if (chosenLibrary >= 0)
			browserEntryActivity.finishAffinity();
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/events/LibraryChosenEventReceiver.kt
package com.lasthopesoftware.bluewater.client.browsing.library.events

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.lasthopesoftware.bluewater.client.browsing.BrowserEntryActivity
import com.lasthopesoftware.bluewater.client.browsing.library.access.session.BrowserLibrarySelection

class LibraryChosenEventReceiver(private val browserEntryActivity: BrowserEntryActivity) : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val chosenLibrary = intent.getIntExtra(BrowserLibrarySelection.chosenLibraryId, -1)
        if (chosenLibrary >= 0) browserEntryActivity.finishAffinity()
    }
}
